hist/histv7/inc/ROOT/RAxis.hxx

```diff
@@ -661,49 +680,30 @@ public:
       auto iNotLess = std::lower_bound(bBegin, bEnd, x);
       int rawbin = iNotLess - bBegin;
       // No need for AdjustOverflowBinNumber(rawbin) here; lower_bound() is the
-      // answer: e.g. for x < *bBegin, rawbin is 0.
+      // answer: e.g. for x < *bBegin, rawbin is -1.
+      if (rawbin < GetFirstBin())
+         return kUnderflowBin;
+      if (rawbin >= GetLastBin() + 1)
+         return kOverflowBin;
       return rawbin;
    }
 
    /// Get the bin center of the bin with the given index.
-   ///
-   /// For the bin at index 0 (i.e. the underflow bin), a bin center of
-   /// `std::numeric_limits<double>::lowest()` is returned, i.e. the smallest
-   /// value that can be held in a double.
-   /// Similarly, for the bin at index N + 1 (i.e. the overflow bin), a bin
-   /// center of `std::numeric_limits<double>::max()` is returned, i.e. the
-   /// largest value that can be held in a double.
-   double GetBinCenter(int bin) const final override
-   {
-      if (IsUnderflowBin(bin))
-         return std::numeric_limits<double>::lowest();
-      if (IsOverflowBin(bin))
-         return std::numeric_limits<double>::max();
-      return 0.5 * (fBinBorders[bin - 1] + fBinBorders[bin]);
-   }
+   /// The result of this method on an overflow or underflow bin is unspecified.
```

Comment:
"Unspecified" is not too helpful for users. We could:
- option1: return invalid numbers or crash
- option2: throw
- option3: return `numeric_limits`
- option4: assert
- option5: `R__ERROR_HERE`.

The current code does option 1. I expect that this will cause trouble in the future... Is option 3 really too expensive?

Reply:
I actually removed option 3 in this case because it wasn't consistent with the behavior of the other types of axes. I however agree that we could implement one of these options to avoid undefined behavior.